test/esjs.spec.js

```diff
@@ -74,6 +79,18 @@ describe('.new()', () => {
     });
   });
 
+  context('given an unsupported analyzer', () => {
```

Comment:
I wonder if it's better to throw this error in the constructor rather than later on.  It would be more helpful to the developer to throw an 'unsupported analyzer' error on the line where the configuration goes in rather than later on.

Reply:
I tend to agree with this, but didn't see any config validation mechanism, and wanted to limit the scope of change. Happy to add in if you'd like.